crates/cheatcodes/src/lib.rs

```diff
@@ -82,6 +83,12 @@ pub(crate) trait Cheatcode: CheatcodeDef + DynCheatcode {
         ccx: &mut CheatsCtxt<DB>,
         executor: &mut E,
     ) -> Result {
+        if self.status() == Status::Deprecated {
+            error!(
+                "{} cheatcode is being deprecated and will be removed in future versions.",
+                self.signature()
+            );
```

Comment:
It may be a bit confusing to have an error be non-throwing but I can see why we would want this to be visible at default verbosity

Reply:
yeah, that will be accomplished with new approach, see https://github.com/foundry-rs/foundry/pull/8883#issuecomment-2357995538